storage-node-v2/src/commands/server.ts

```diff
@@ -29,34 +40,177 @@ export default class Server extends ApiCommandBase {
+        account,
+        workerId,
+        maxFileSize,
+        uploadsDir: flags.uploads,
+        tempDirName: TempDirName,
+        process: this.config,
+        queryNodeUrl,
+        enableUploadingAuth: !flags.disableUploadAuth,
+        elasticSearchEndpoint: elasticUrl,
+      })
       logger.info(`Listening on http://localhost:${port}`)
       app.listen(port)
     } catch (err) {
-      logger.error(`Error: ${err}`)
+      logger.error(`Server error: ${err}`)
     }
   }
 
   // Override exiting.
   /* eslint-disable @typescript-eslint/no-empty-function */
   async finally(): Promise<void> {}
 }
+
+/**
+ * Run the data syncronization process.
+ *
+ * @param workerId - worker ID
+ * @param queryNodeUrl - Query Node for data fetching
+ * @param uploadsDir - data uploading directory
+ * @param syncWorkersNumber - defines a number of the async processes for sync
+ * @param syncIntervalMinutes - defines an interval between sync runs
+ *
+ * @returns void promise.
+ */
+function runSyncWithInterval(
+  workerId: number,
+  queryNodeUrl: string,
+  uploadsDirectory: string,
+  syncWorkersNumber: number,
+  syncIntervalMinutes: number
+) {
+  setTimeout(async () => {
+    const sleepIntevalInSeconds = syncIntervalMinutes * 60 * 1000
+
+    logger.info(`Sync paused for ${syncIntervalMinutes} minute(s).`)
+    await sleep(sleepIntevalInSeconds)
+    logger.info(`Resume syncing....`)
+
+    try {
+      await performSync(workerId, syncWorkersNumber, queryNodeUrl, uploadsDirectory)
+    } catch (err) {
+      logger.error(`Critical sync error: ${err}`)
+    }
+
+    runSyncWithInterval(workerId, queryNodeUrl, uploadsDirectory, syncWorkersNumber, syncIntervalMinutes)
+  }, 0)
+}
+
+/**
+ * Removes the temporary directory from the uploading directory.
+ * All files in the temp directory are deleted.
+ *
+ * @param uploadsDir - data uploading directory
+ * @param tempDirName - temporary directory name within the uploading directory
+ * @returns void promise.
+ */
+async function removeTempDirectory(uploadsDir: string, tempDirName: string): Promise<void> {
+  try {
+    logger.info(`Removing temp directory ...`)
+    const tempFileUploadingDir = path.join(uploadsDir, tempDirName)
+
+    const rimrafAsync = promisify(rimraf)
+    await rimrafAsync(tempFileUploadingDir)
+  } catch (err) {
+    logger.error(`Removing temp directory error: ${err}`)
+  }
+}
+
+/**
+ * Verifies the worker ID from the command line argument and provided Joystream account.
+ * It throws an error when not matched.
+ *
+ * @param api - runtime API promise
+ * @param workerId - worker ID from the command line arguments
+ * @param account - Joystream account KeyringPair
+ * @returns void promise.
+ */
+async function verifyWorkerId(api: ApiPromise, workerId: number, account: KeyringPair): Promise<void> {
+  // Cast Codec type to Worker type
+  const workerObj = (await api.query.storageWorkingGroup.workerById(workerId)) as unknown
+  const worker = workerObj as Worker
```

Comment:
Since you're using api augmentations this will be sufficient:
```suggestion
  const worker = await api.query.storageWorkingGroup.workerById(workerId)
```

Reply:
You are correct. Fixed here: https://github.com/Joystream/joystream/pull/2686/commits/7230f23e8e30966c0cac51a17f528926f6069d0b